### PR TITLE
[JOY] Fix missing UI elements and French translation

### PR DESCRIPTION
**Pull request by KyleKatarn31**
## Purpose

- Some UI elements are missing in current implementation
- Some UI elements are not displayed correctly in French

game1
game2
game3

## Proposed changes

- Missing elements added
- UI elements corrected in FR-fr translation

game4
game5
game6
i

### CODE REVIEW

File: dll/cpl/joy/joy.c [hunk 1]
```diff
@@ -80,8 +80,17 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
     switch (uMsg)
     {
         case WM_INITDIALOG:
+        {
+            WCHAR szBuf[256];
+            HWND dwComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
+            if (dwComboHwnd)
+            {
+                LoadString(hApplet, IDS_NONE, szBuf, _countof(szBuf));
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:88]: LoadStringW

File: dll/cpl/joy/joy.c [hunk 2]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
     UNREFERENCED_PARAMETER(lParam);
 
+    WCHAR szBuf[2];
+    HWND dwComboHwnd;
+
     switch (uMsg)
     {
         case WM_INITDIALOG:
+            szBuf[1]=UNICODE_NULL;
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:133]: Add spaces around =

File: dll/cpl/joy/joy.c [hunk 3]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+
+            CheckDlgButton(hwndDlg, IDC_JOYSTICK_RADIO, BST_CHECKED);
+
+            dwComboHwnd = GetDlgItem(hwndDlg,IDC_AXES_COMBO);
+            if (dwComboHwnd)
+            {
+                szBuf[0]=L'2';
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, -1, (LPARAM)szBuf);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:141]: What's this -1? It can be zero.

File: dll/cpl/joy/joy.c [hunk 4]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
     {
         case WM_INITDIALOG:
+            szBuf[1]=UNICODE_NULL;
+
+            CheckDlgButton(hwndDlg, IDC_JOYSTICK_RADIO, BST_CHECKED);
+
+            dwComboHwnd = GetDlgItem(hwndDlg,IDC_AXES_COMBO);
+            if (dwComboHwnd)
+            {
+                szBuf[0]=L'2';
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:140]: spaces around =

File: dll/cpl/joy/joy.c [hunk 5]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, -1, (LPARAM)szBuf);
+                szBuf[0]=L'3';
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, -1, (LPARAM)szBuf);
+                szBuf[0]=L'4';
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, -1, (LPARAM)szBuf);
+                SendMessageW(dwComboHwnd, CB_SETCURSEL , 0, (LPARAM)NULL);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:146]: (LPARAM)NULL is zero

**Comment by KyleKatarn31** [comment]
@binarymaster : "enhancement" to be added (for the IU improvement and initialization, regardless of the language used)

File: dll/cpl/joy/joy.c (hunk 1 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:88]: ok

File: dll/cpl/joy/joy.c (hunk 2 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:133]: ok

File: dll/cpl/joy/joy.c (hunk 3 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:141]: Yes (copy / paste from another ROS module where -1 was used. Likely to be cleanup up there someday)

File: dll/cpl/joy/joy.c (hunk 4 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:140]: ok

File: dll/cpl/joy/joy.c (hunk 5 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:146]: ok

File: dll/cpl/joy/joy.c [hunk 6]
```diff
@@ -80,8 +80,17 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
     switch (uMsg)
     {
         case WM_INITDIALOG:
+        {
+            WCHAR szBuf[256];
+            HWND dwComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
+            if (dwComboHwnd)
+            {
+                LoadStringW(hApplet, IDS_NONE, szBuf, _countof(szBuf));
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, -1, (LPARAM)szBuf);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c]: -1 to 0

File: dll/cpl/joy/joy.c [hunk 7]
```diff
@@ -80,8 +80,17 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
     switch (uMsg)
     {
         case WM_INITDIALOG:
+        {
+            WCHAR szBuf[256];
+            HWND dwComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:85]: Wrong Hungarian notation

File: dll/cpl/joy/joy.c [hunk 8]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
     UNREFERENCED_PARAMETER(lParam);
 
+    WCHAR szBuf[2];
+    HWND dwComboHwnd;
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:128]: Wrong Hungarian notation

File: dll/cpl/joy/joy.c [hunk 9]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+
     switch (uMsg)
     {
         case WM_INITDIALOG:
+            szBuf[1] = UNICODE_NULL;
+
+            CheckDlgButton(hwndDlg, IDC_JOYSTICK_RADIO, BST_CHECKED);
+
+            dwComboHwnd = GetDlgItem(hwndDlg,IDC_AXES_COMBO);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:137]: insert space after comma

File: dll/cpl/joy/joy.c [hunk 10]
```diff
@@ -23,12 +23,12 @@
  * PURPOSE:         ReactOS Software Control Panel
  * PROGRAMMER:      Dmitry Chapyshev (lentind@yandex.ru)
  * UPDATE HISTORY:
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:25]: Add your name here

File: dll/cpl/joy/joy.c [hunk 11]
```diff
@@ -80,8 +80,17 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
     {
         case WM_INITDIALOG:
+        {
+            WCHAR szBuf[256];
+            HWND dwComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
+            if (dwComboHwnd)
+            {
+                LoadStringW(hApplet, IDS_NONE, szBuf, _countof(szBuf));
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+                SendMessageW(dwComboHwnd, CB_SETCURSEL , 0, (LPARAM)NULL);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:90]: Delete space before comma

File: dll/cpl/joy/joy.c [hunk 12]
```diff
@@ -115,9 +124,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+                szBuf[0] = L'4';
+                SendMessageW(dwComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+                SendMessageW(dwComboHwnd, CB_SETCURSEL , 0, 0);
+            }
+
+            dwComboHwnd = GetDlgItem(hwndDlg,IDC_BUTTONS_COMBO);
+            if (dwComboHwnd)
+            {
+                szBuf[0] = L'0';
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:152]: Why don't use loop?

File: dll/cpl/joy/joy.c (hunk 6 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: Done

File: dll/cpl/joy/joy.c (hunk 7 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:85]: OK

File: dll/cpl/joy/joy.c (hunk 8 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:128]: ok

File: dll/cpl/joy/joy.c (hunk 9 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:137]: ok

File: dll/cpl/joy/joy.c (hunk 10 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:25]: Content yes, name no (when i'll be a dev here :-) )

File: dll/cpl/joy/joy.c (hunk 11 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:90]: ok

File: dll/cpl/joy/joy.c (hunk 12 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:152]: Too few steps, no positive impact on code efficiency or maintainability.

**Comment by KyleKatarn31** [comment]
Thank you @katahiromz . Ready for round 3 ! :)

File: dll/cpl/joy/joy.c [hunk 13]
```diff
@@ -80,8 +81,17 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
     switch (uMsg)
     {
         case WM_INITDIALOG:
+        {
+            WCHAR szBuf[256];
+            HWND hComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:86]: insert space after comma

File: dll/cpl/joy/joy.c [hunk 14]
```diff
@@ -115,9 +125,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
     UNREFERENCED_PARAMETER(lParam);
 
+    WCHAR szBuf[2];
+    HWND hComboHwnd;
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:129]: Move these two lines above UNREFERENCED_PARAMETER(lParam);

File: dll/cpl/joy/joy.c [hunk 15]
```diff
@@ -115,9 +125,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+
     switch (uMsg)
     {
         case WM_INITDIALOG:
+            szBuf[1] = UNICODE_NULL;
+
+            CheckDlgButton(hwndDlg, IDC_JOYSTICK_RADIO, BST_CHECKED);
+
+            hComboHwnd = GetDlgItem(hwndDlg, IDC_AXES_COMBO);
+            if (hComboHwnd)
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:139]: excessive NULL check

File: dll/cpl/joy/joy.c [hunk 16]
```diff
@@ -115,9 +125,43 @@ CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+                SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+                szBuf[0] = L'3';
+                SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+                szBuf[0] = L'4';
+                SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+                SendMessageW(hComboHwnd, CB_SETCURSEL, 0, 0);
+            }
+
+            hComboHwnd = GetDlgItem(hwndDlg, IDC_BUTTONS_COMBO);
+            if (hComboHwnd)
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:151]: excessive NULL check

File: dll/cpl/joy/joy.c (hunk 13 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:86]: ok

File: dll/cpl/joy/joy.c (hunk 14 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:129]: ok

File: dll/cpl/joy/joy.c (hunk 15 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:139]: ok

File: dll/cpl/joy/joy.c (hunk 16 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:151]: ok

**Comment by KyleKatarn31** [comment]
Thank you @katahiromz . Ready for round 4 ! :)

File: dll/cpl/joy/joy.c [hunk 17]
```diff
@@ -80,8 +81,17 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
     switch (uMsg)
     {
         case WM_INITDIALOG:
+        {
+            WCHAR szBuf[256];
+            HWND hComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
+            if (hComboHwnd)
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:87]: excessive NULL check

Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: fixed

**Comment by KyleKatarn31** [comment]
Thank you @katahiromz . Ready for round 5 or Approval :)

**Comment by katahiromz** [comment]
Wait two days for cooling down. If something is wrong, somebody will claim maybe.

**Comment by KyleKatarn31** [comment]
I agree on the "2 days low-pass filter" approach before moving forward :-)

File: dll/cpl/joy/joy.c [hunk 18]
```diff
@@ -75,11 +76,19 @@ AddColumns(HWND hList)
...
+    WCHAR szBuf[256];
+    HWND hComboHwnd;
+
     UNREFERENCED_PARAMETER(lParam);
 
     switch (uMsg)
     {
         case WM_INITDIALOG:
+            hComboHwnd = GetDlgItem(hwndDlg,IDC_PREFERRED_DEV_COMBO);
+
```
Comment (HBelusca) [inline dll/cpl/joy/joy.c:88]:
```suggestion
        {
            WCHAR szBuf[256];
            HWND hComboHwnd = GetDlgItem(hwndDlg, IDC_PREFERRED_DEV_COMBO);
```

File: dll/cpl/joy/joy.c [hunk 19]
```diff
@@ -75,11 +76,19 @@ AddColumns(HWND hList)
...
+            LoadStringW(hApplet, IDS_NONE, szBuf, _countof(szBuf));
+            SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+            SendMessageW(hComboHwnd, CB_SETCURSEL, 0, (LPARAM)NULL);
             break;
```
Comment (HBelusca) [inline dll/cpl/joy/joy.c:92]:
```suggestion
            break;
        }
```

File: dll/cpl/joy/joy.c [hunk 20]
```diff
@@ -75,11 +76,19 @@ AddColumns(HWND hList)
 INT_PTR CALLBACK
 AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
+    WCHAR szBuf[256];
+    HWND hComboHwnd;
+
```
Comment (HBelusca) [inline dll/cpl/joy/joy.c:81]:
```suggestion
```

File: dll/cpl/joy/joy.c [hunk 21]
```diff
@@ -113,11 +122,40 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
 {
+    WCHAR szBuf[2];
+    HWND hComboHwnd;
+
     UNREFERENCED_PARAMETER(lParam);
 
     switch (uMsg)
     {
         case WM_INITDIALOG:
+            szBuf[1] = UNICODE_NULL;
```
Comment (HBelusca) [inline dll/cpl/joy/joy.c:133]:
```suggestion
        {
            HWND hComboHwnd;
            WCHAR szBuf[2] = {0, 0};
```

File: dll/cpl/joy/joy.c [hunk 22]
```diff
@@ -113,11 +122,40 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 INT_PTR CALLBACK
 CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
+    WCHAR szBuf[2];
+    HWND hComboHwnd;
+
```
Comment (HBelusca) [inline dll/cpl/joy/joy.c:127]:
```suggestion
```

File: dll/cpl/joy/joy.c [hunk 23]
```diff
@@ -113,11 +122,40 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
+            SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+            szBuf[0] = L'2';
+            SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+            szBuf[0] = L'3';
+            SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+            szBuf[0] = L'4';
+            SendMessageW(hComboHwnd, CB_ADDSTRING, 0, (LPARAM)szBuf);
+            SendMessageW(hComboHwnd, CB_SETCURSEL, 4, 0);
+
             break;
```
Comment (HBelusca) [inline dll/cpl/joy/joy.c:159]:
```suggestion
            break;
        }
```

Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:162]: ok

File: dll/cpl/joy/joy.c (hunk 21 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:136]: I prefer to keep UNICODE_NULL but I take the other part

File: dll/cpl/joy/joy.c (hunk 22 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: ok

File: dll/cpl/joy/joy.c (hunk 20 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: ok

File: dll/cpl/joy/joy.c (hunk 18 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: ok

File: dll/cpl/joy/joy.c (hunk 19 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c:92]: ok

**Comment by KyleKatarn31** [comment]
My mistake.... I need to rollback my last force-push

**Comment by KyleKatarn31** [comment]
Thank you @ThFabba for fixing the undue review request following my messy force-push
@HBelusca : I let you do a last sanity check but it should now be 100% in line with your comments.

OK for merge ?

File: dll/cpl/joy/joy.c [hunk 24]
```diff
@@ -75,12 +76,21 @@ AddColumns(HWND hList)
 INT_PTR CALLBACK
 AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
+
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:79]: Delete this empty line

File: dll/cpl/joy/joy.c [hunk 25]
```diff
@@ -113,12 +123,43 @@ AdvancedPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 INT_PTR CALLBACK
 CustomPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
+
```
Comment (katahiromz) [inline dll/cpl/joy/joy.c:126]: Delete this empty line

File: dll/cpl/joy/joy.c (hunk 24 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: ok

File: dll/cpl/joy/joy.c (hunk 25 above)
Comment (KyleKatarn31) [reply dll/cpl/joy/joy.c]: ok

**Comment by KyleKatarn31** [comment]
Thank you @katahiromz : Comments taken into account.

**Pull request merged**